openmc/deplete/results_list.py

```diff
@@ -192,3 +194,78 @@ def get_depletion_time(self):
         for ix, res in enumerate(items):
             times[ix] = res.proc_time
         return times
+
+    def export_to_materials(self, burnup_index, nuc_with_data=None):
+        """Converts a depletion time point composition into an openmc.Materials
+        instance, useful for running transport calculations on individual
+        depletion time points
+
+        This requires a materials.xml file to be present in the current
+        directory. This should correspond to the materials initially
+        present in the depletion calculation.
+
+        Only materials found in the original materials.xml file will be
+        updated. If for some reason you have modified OpenMC to produce
+        new materials as depletion takes place, this method will not
+        work as expected and leave out that material.
+
+        Parameters
+        ----------
+        burn_index : int
+            Index of burnup step to evaluate
+        nuc_with_data : Iterable of str, optional
+            nuclides iterable to evaluate with neutron data
+
+            This must be specified because not all nuclides appearing in
+            depletion results have associated neutron cross sections, and
+            as such cannot be used in subsequent transport calculations.
+            If not provided, nuclides from the cross_sections element of
+            materials.xml will be used. If that element is not present,
+            nuclides from OPENMC_CROSS_SECTIONS will be used.
+        Returns
+        -------
+        mat_file : Materials
+            A modified Materials instance containing depleted material data,
+            plus original isotopic compositions of non-depletable materials
+        """
+        result = self[burnup_index]
+        mat_file = openmc.Materials.from_xml("materials.xml")
+
+        # Only nuclides with valid transport data will be written to
+        # the new materials XML file. The precedence of nuclides to select
+        # is first ones provided as a kwarg here, then ones specified
+        # in the materials.xml file if provided, then finally from
+        # the environment variable OPENMC_CROSS_SECTIONS.
+        if nuc_with_data:
+            if not all(isinstance(nuclide_name, str) for nuclide_name in nuc_with_data):
+                raise InvalidArgumentError("Expected an iterable of strings as acceptable nuclide data.")
```

Comment:
Non-blocking, but we have `openmc.checkvalue.check_iterable_type https://github.com/openmc-dev/openmc/blob/18af12925a99866b6d07e6769ac2346b4c5085ba/openmc/checkvalue.py#L63
that can handle this for you